File: wails/model/book.go
package model

type ProgressType string

const (
	ProgressPage       ProgressType = "page"
	ProgressLocation   ProgressType = "location"
	ProgressPercentage ProgressType = "percentage"
)

type Book struct {
	ID              string  `json:"id"`
	Title           string  `json:"title"`
	Author          *string `json:"author,omitempty"`
	CurrentProgress int     `json:"current_progress"`
	TotalProgress   int     `json:"total_progress"`
	ProgressType    string  `json:"progress_type"`
	TargetDate      *string `json:"target_date,omitempty"`
	CompletedAt     *string `json:"completed_at,omitempty"`
	CreatedAt       string  `json:"created_at"`
}

type CreateBookRequest struct {
	Title         string  `json:"title"`
	Author        *string `json:"author,omitempty"`
	TotalProgress int     `json:"total_progress"`
	ProgressType  string  `json:"progress_type"`
	TargetDate    *string `json:"target_date,omitempty"`
}

type UpdateBookRequest struct {
	ID              string  `json:"id"`
	Title           *string `json:"title,omitempty"`
	Author          *string `json:"author,omitempty"`
	CurrentProgress *int    `json:"current_progress,omitempty"`
	TotalProgress   *int    `json:"total_progress,omitempty"`
	ProgressType    *string `json:"progress_type,omitempty"`
	TargetDate      *string `json:"target_date,omitempty"`
}

File: deprecated/src-tauri/src/models/book.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Book {
    pub id: String,
    pub title: String,
    pub author: Option<String>,
    pub current_progress: i32,
    pub total_progress: i32,
    pub progress_type: String,
    pub target_date: Option<String>,
    pub completed_at: Option<String>,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateBookRequest {
    pub title: String,
    pub author: Option<String>,
    pub total_progress: i32,
    pub progress_type: String,
    pub target_date: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpdateBookRequest {
    pub id: String,
    pub title: Option<String>,
    pub author: Option<String>,
    pub current_progress: Option<i32>,
    pub total_progress: Option<i32>,
    pub progress_type: Option<String>,
    pub target_date: Option<String>,
}
